test/e2e/validation_test.go

```diff
@@ -46,6 +46,39 @@ type winService struct {
 	description string
 }
 
+// testKubeletPriority tests if kubelet priority class is set to "AboveNormal"
+func testKubeletPriority(t *testing.T) {
+	tc, err := NewTestContext()
+	require.NoError(t, err)
+	requiredPriorityClass := "AboveNormal"
+
+	t.Run("nodes", func(t *testing.T) {
```

Comment:
consider moving this into the for loop so a separate test is executed for each node (and you can name each test with the node name)